src/LiveComponent/src/Metadata/LiveComponentMetadataFactory.php

```diff
@@ -59,32 +59,38 @@ public function createPropMetadatas(\ReflectionClass $class): array
                 continue;
             }
 
-            $collectionValueType = null;
-            $infoTypes = $this->propertyTypeExtractor->getTypes($class->getName(), $property->getName()) ?? [];
-            foreach ($infoTypes as $infoType) {
-                if ($infoType->isCollection()) {
-                    foreach ($infoType->getCollectionValueTypes() as $valueType) {
-                        $collectionValueType = $valueType;
-                        break;
-                    }
+            $metadatas[$property->getName()] = $this->createLivePropMetadata($class->getName(), $property->getName(), $property, $attribute);
+        }
+
+        return array_values($metadatas);
+    }
+
+    public function createLivePropMetadata(string $className, string $propertyName, \ReflectionProperty $property, \ReflectionAttribute $attribute = null): LivePropMetadata
```

Comment:
Minor: let's make `\ReflectionAttribute` argument a `LiveProp` argument and make it required. The method above can call `$attribute->newInstance()` and `LiveComponentHydrator` can call `new LiveProp()`.